pkg/operator/controller/ingress/rsyslog_configmap.go

```diff
@@ -42,26 +42,31 @@ func (r *reconciler) ensureRsyslogConfigMap(ic *operatorv1.IngressController, de
 	case !wantCM && !haveCM:
 		return false, nil, nil
 	case !wantCM && haveCM:
-		if deleted, err := r.deleteRsyslogConfigMap(current); err != nil {
-			return true, current, fmt.Errorf("failed to delete configmap: %v", err)
-		} else if deleted {
+		if err := r.client.Delete(context.TODO(), current); err != nil {
+			if !errors.IsNotFound(err) {
+				return true, current, fmt.Errorf("failed to delete configmap: %v", err)
+			}
+		} else {
 			log.Info("deleted configmap", "configmap", current)
 		}
+		return false, nil, nil
 	case wantCM && !haveCM:
 		if created, err := r.createRsyslogConfigMap(desired); err != nil {
 			return false, nil, fmt.Errorf("failed to create configmap: %v", err)
 		} else if created {
 			log.Info("created configmap", "configmap", desired)
+			return r.currentRsyslogConfigMap(ic)
 		}
```

Comment:
Similarly to the case I commented on in `ensureServiceMonitor`, if `err == nil && !created`, control flows down to the `return true, current, nil` statement below, and `current` is nil.